File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.js
function solution(record) {

    const map = new Map();

    const splitRecord = record.map(r => r.split(" "));

    // 닉네임 최신화
    for (const r of splitRecord) {
        if (r[0] === "Enter" || r[0] === "Change") {
            map.set(r[1], r[2]);
        }
    }

    const result = [];

    // 결과 생성
    for (const r of splitRecord) {

        switch (r[0]) {

            case "Enter":
                result.push(`${map.get(r[1])}님이 들어왔습니다.`);
                break;

            case "Leave":
                result.push(`${map.get(r[1])}님이 나갔습니다.`);
                break;
        }
    }

    return result;
}

File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.java
import java.util.*;
import java.util.stream.Collectors;

class Solution {
    public String[] solution(String[] record) {
        Map<String, String> map = new HashMap<>();
        List<String[]> splitRecord = Arrays.stream(record).map(r -> r.split(" ")).collect(Collectors.toList());

        for (String[] r : splitRecord) {
            if (r[0].equals("Enter") || r[0].equals("Change")) {
                map.put(r[1], r[2]);
            }
        }

        List<String> result = new ArrayList<>();

        for (String[] r : splitRecord) {
            switch (r[0]) {
                case "Enter":
                    result.add(String.format("%s님이 들어왔습니다.", map.get(r[1])));
                    break;
                case "Leave":
                    result.add(String.format("%s님이 나갔습니다.", map.get(r[1])));
                    break;
            }
        }
        return result.toArray(new String[0]);
    }
}